packages/application-shell/TEST_UTILS.md

```diff
@@ -0,0 +1,376 @@
+  });
+});
+```
+
+Here we overwrite the authenticated user's `firstName` for our test. The data we pass in gets merged with the default data.
+
+We can also test the case in which no user is authenticated by passing
+
+```jsx
+import { render } from '@commercetools-frontend/application-shell/test-utils';
+
+describe('FirstName', () => {
+  it('should render the authenticated users first name', () => {
+    const { container, user } = render(<FirstName />, { user: null });
+    expect(container).toHaveTextContent('Anonymous');
+  });
+});
+```
+
+### API
+
+This section describes the methods exported by `@commercetools-frontend/application-shell/test-utils`.
+
+`test-utils` is builds on top of `react-testing-library`, so all `react-testing-library` exports are available. It should not be necessary for you to import `react-testing-library` into your tests. The following section describes the additional exports added on top of `react-testing-library` and the overwritten `render` of `react-testing-library` itself.
+
+#### `render(ui: ReactElement, options: Object)`
+
+| Argument              | Type          | Concern           | Description                                                                                                                                                                                                                                                                                                                                                                                                        |
+| --------------------- | ------------- | ----------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `ui`                  | React Element | React             | React Element to render.                                                                                                                                                                                                                                                                                                                                                                                           |
+| `options.locale`      | `String`      | Localisation      | Determines the UI language and number format, is used to configure `IntlProvider`. Only _core_ messages will be available during tests, no matter the `locale`.                                                                                                                                                                                                                                                    |
```

Comment:
Maybe we should have a second document which explains `locale` and  `dataLocale`, which we could link to from here?

Reply:
I think it's fine to just shorty explain it here. We don't need to write an essay 😉

Reply:
I would rather write an essay though because "`locale` vs `dataLocale`" has lead to a ton of confusion and the only proper documentation around the whole languages stuff is a comment on a GitHub issue in a private repo.

Reply:
We can have those documents yes, where we can link to. I still think that a short TL;DR here is also useful.

But I don't want to block this, we can iterate on this.